src/Provider/Photon/Photon.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Photon;
+
+use Geocoder\Collection;
+use Geocoder\Exception\InvalidServerResponse;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Location;
+use Geocoder\Model\AddressBuilder;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Provider\Provider;
+use Geocoder\Provider\Photon\Model\PhotonAddress;
+use Http\Client\HttpClient;
+
+/**
+ * @author Niklas Närhinen <niklas@narhinen.net>
+ * @author Jonathan Beliën <jbe@geo6.be>
+ */
+final class Photon extends AbstractHttpProvider implements Provider
+{
+    /**
+     * @var string
+     */
+    private $rootUrl;
+
+    /**
+     * @param HttpClient $client an HTTP client
+     *
+     * @return Photon
+     */
+    public static function withKomootServer(HttpClient $client)
```

Comment:
Maybe add default value for  $rootUrl in `__construct`?
 ` __construct(HttpClient $client, $rootUrl = 'https://photon.komoot.de')`

Reply:
That could be an option indeed !
I don't really have an opinion on this but I like the fact that by using `Photon::withKomootServer()` the developer knows for a fact that he will use server from Komoot (instead of its own for instance).

Reply:
Why you don't add return type declaration for this method `withKomootServer(HttpClient $client): self` ?

Reply:
That's indeed missing 👍